Deque

satyampgt4 opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.

Help us understand your motivation by explaining why you decided to make this change.
You can learn more about contributing to NeoAlgo here: https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md

Happy Contributing!

-->

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

It is an implementation of Deque using doubly linked list in c++.


### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

 #1602

satyampgt4 committed:
Add files via upload

satyampgt4 committed:
Update README.md

satyampgt4 committed:
update README.md

satyampgt4 committed:
Merge branch 'master' into deque

HarshCasper commented on C-Plus-Plus/ds/Deque.cpp:
```diff
@@ -0,0 +1,215 @@
+
+    cout<<"size of deque :"<<q.size_deque()<<endl;
+
+    q.enqueue_front(25);
+    // q will be : 25 15 5 6
+    q.enqueue_back(30);
+    // q will be : 25 15 5 6 30
+    q.enqueue_back(11);
+    // q will be : 25 15 5 6 30 11
+    q.enqueue_front(35);
+    // q will be : 35 25 15 5 6 30 11
+    q.display();
```
Use Dynamic Input instead of a Driver Code.

satyampgt4 replied:
i use the drive code to show how these function will execute

HarshCasper replied:
We need the Driver Code to help the User use the Code. Not to show how the Code is being executed. If not changed, we will not be able to merge this.

HarshCasper commented on C-Plus-Plus/ds/Deque.cpp:
```diff
@@ -0,0 +1,215 @@
+/************************************************************************
+
+In this Program we will implement Double Ended Queue (de_que)
```
Can you add a beginner pitch about Double Ended Queue?

HarshCasper commented on C-Plus-Plus/ds/Deque.cpp:
```diff
@@ -0,0 +1,215 @@
+
+
+
+
```
Remove unnecessary lines

HarshCasper commented on C-Plus-Plus/ds/Deque.cpp:
```diff
@@ -0,0 +1,215 @@
+    {
+        head = NULL;
+        tail = NULL;
+        n=0;
```
The Formatting is inconsistent. Try to keep it consistent for better uniformity.

satyampgt4 committed:
Update Deque

beginner pitch about Double Ended Queue is added ,Unnecessary lines are removed, Made formatting consistent.

Khimit commented:
@HarshCasper  can you please assign this to me?? from DWoC

HarshCasper commented:
> @HarshCasper can you please assign this to me?? from DWoC

PRs are not meant to be assigned.

HarshCasper commented:
I'm closing the PR due to no activity. If you would like to reopen this PR to be reviewed please join the [Slack](https://join.slack.com/t/tesseractcodi-bxq5968/shared_invite/zt-ju6c0nqq-KgkTvFyxjNVbDRBIjADanw) and drop me a message.
